packages/components/src/components/KeyValueList/KeyValue.test.js

```diff
@@ -0,0 +1,194 @@
+    keyValues: [{ key: 'foo0', value: 'bar0', id: 'aaaa' }],
+    onAdd() {},
+    onRemove() {}
+  };
+  const { getByValue } = render(<KeyValueList {...props} />);
+
+  fireEvent.change(getByValue(/foo0/i), {
+    target: { value: 'new key 0' }
+  });
+
+  expect(props.onChange).toHaveBeenCalledTimes(1);
+  expect(props.onChange).toBeCalledWith({
+    index: 0,
+    type: 'key',
+    value: 'new key 0'
+  });
+});
+
+it('KeyValueList change value', () => {
+  const props = {
+    onChange: jest.fn(),
+    invalidFields: {},
+    keyValues: [{ key: 'foo0', value: 'bar0', id: 'aaaa' }],
+    onAdd() {},
+    onRemove() {}
+  };
+  const { getByValue } = render(<KeyValueList {...props} />);
+
+  fireEvent.change(getByValue(/bar0/i), {
+    target: { value: 'new value 0' }
+  });
+
+  expect(props.onChange).toHaveBeenCalledTimes(1);
+  expect(props.onChange).toBeCalledWith({
+    index: 0,
+    type: 'value',
+    value: 'new value 0'
+  });
+});
+
+it('KeyValueList delete button works', () => {
+  const props = {
+    onChange: jest.fn(),
+    invalidFields: {},
+    keyValues: [
+      { key: 'foo0', value: 'bar0', id: 'aaaa' },
+      { key: 'foo1', value: '', id: 'bbbb' },
+      { key: 'foo2', value: 'bar2', id: 'cccc' },
+      { key: 'foo3', value: 'bar3', id: 'dddd' }
+    ],
+    deleteButton: true,
+    onAdd() {},
+    onRemove() {}
+  };
+  const { getByLabelText } = render(<KeyValueList {...props} />);
+
+  const deleteButton0 = getByLabelText(/Delete Button#0/i);
+
+  fireEvent.click(deleteButton0);
+
+  expect(props.onChange).toHaveBeenCalledTimes(1);
+  expect(props.onChange).toBeCalledWith({
+    index: 0,
+    type: 'value',
+    value: ''
+  });
+});
+
+it('KeyValueList delete button hidden', () => {
+  const props = {
+    onChange: jest.fn(),
+    invalidFields: {},
+    keyValues: [{ key: 'foo0', value: 'bar0', id: 'aaaa' }],
+    onAdd() {},
+    onRemove() {}
+  };
+  const { queryByLabelText } = render(<KeyValueList {...props} />);
+
+  expect(queryByLabelText(/Delete Button/i)).toBeFalsy();
+});
+
+it('KeyValueList add and remove buttons work', () => {
+  const props = {
+    onChange() {},
+    invalidFields: {},
+    keyValues: [
+      { key: 'foo0', value: 'bar0', id: 'aaaa' },
+      { key: 'foo1', value: '', id: 'bbbb' },
+      { key: 'foo2', value: 'bar2', id: 'cccc' },
+      { key: 'foo3', value: 'bar3', id: 'dddd' }
+    ],
+    onAdd: jest.fn(),
+    onRemove: jest.fn()
+  };
+  const { getByLabelText } = render(<KeyValueList {...props} />);
+
+  const removeButton = getByLabelText(/Remove Button/i);
+  const addButton = getByLabelText(/Add Button/i);
+
+  fireEvent.click(addButton);
+  fireEvent.click(addButton);
+  fireEvent.click(addButton);
+  fireEvent.click(addButton);
+  fireEvent.click(addButton);
+
+  fireEvent.click(removeButton);
+  fireEvent.click(removeButton);
+  fireEvent.click(removeButton);
+
+  expect(props.onAdd).toHaveBeenCalledTimes(5);
+  expect(props.onRemove).toHaveBeenCalledTimes(3);
```

Comment:
With the current implementation, we don't care about the values because you add/remove to/from the list of KeyValues like it's a stack. There is no index or important information being passed in the `onAdd` or `onRemove` callback functions.